Sources/SwiftParser/Lookahead.swift

```diff
@@ -90,6 +91,7 @@ extension Parser.Lookahead {
   mutating func consumeAnyToken() {
     tokensConsumed += 1
     self.currentToken = self.lexemes.advance()
+    self.lexemes.recordNextTokenInLookaheadTracker()
```

Comment:
As the second part of my comment https://github.com/apple/swift-syntax/pull/1685#discussion_r1254507635:

Could we also call `recordNextTokenInLookaheadTracker` from `LexemeSequence.advance` so that we never need to call it from `Lookahead` or `Parser` (and could maybe even make that method `private`)? Or is there a performance regression associated with it because we’d now effectively also be calling `recordNextTokenInLookaheadTracker` from `Parser.consumeAnyToken`.

In either case, I think we should be consistent about who’s responsible for calling `recordNextTokenInLookaheadTracker`. Either `LexemeSequence` calls it from both `advance` and `peek` or `Lookahead` and `Parser` are responsible for calling it. I don’t think we should end up in a situation where `LexemeSequence` calls it on `peek` but `Lookeahead`/`Parser` need to call it on `advance` because that sort of inconsistency is a source of bugs.

Reply:
Oops, I think that's something I forgot to delete after doing the performance test. `recordNextTokenInLookaheadTracker` is already added to `LexemeSequence.advance`.